content/blog/2020-08-14-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,242 @@
+
+```{r,message=FALSE}
+library(scales)
+
+left_over %>%
+  ggplot(aes(quarter, left_over)) +
+  geom_point() +
+  geom_line(group = 1) +
+  facet_wrap(~expense) +
+  scale_y_continuous(labels = dollar) +
+  geom_hline(yintercept = 20, color = "red") +
+  labs(
+    title = "Spending by expense type",
+    x = "Quarter",
+    y = "Amount jeft over, in $"
+  )
+```
+
+Voila! We have both a summary table and a graphic that can tell us useful information about company spending. Hmm, there might be something to the trend we see with the overspending on happy hours in Q3 after Q1 and Q2 were sunk with lawsuits.
+
+
+## Join the party
+
+We don't always have all of the information we need in a single data file and many times we need to bring in data from other sources to supplement and augment our existing data in the course of an analysis.  
+
+For our second scenario, we'll leave the finance world and move on to something that is perhaps a little more exciting: superheroes!
+Below is a data set containing information on various comic book characters.  
+Over time this data set has been expanded on with multiple sheets including new information. 
+Before we start exploring, we'll need to join the relevant data from the two sheets so we can conduct some fun analyses.
+
+First let's load our data. 
+Our data set contains two sheets so we'll load them in separately as two distinct data frames.
+
+If you would like to follow along, you can download the Excel file called `superheroes.xlsx` [here](https://drive.google.com/file/d/1KAOAMA_iRdMrl0erYd2DTm7uUKCsDuM9/view?usp=sharing) and remember that you can also follow along with the analysis on RStudio Cloud [here](https://rstudio.cloud/project/1523407). 
+
+```{r}
+superheroes <- read_excel("data/superheroes.xlsx", sheet = "heroes")
+identities <- read_excel("data/superheroes.xlsx", sheet = "alter_egos")
+```
+
+Let's take a look at the sheet with data on superheroes.
+
+```{r}
+superheroes
+```
+
+And now let's take a look at the sheet listing the alter ego of each character.
+
+```{r}
+identities
+```
+
+In each sheet, there is a `hero` column that allows us to match data from one sheet to the other.   
+Using `left_join()` from the dplyr package, we can bring in the data from the identities data frame into our superheroes data frame.
+
+To do this we'll call `left_join()` on superheroes, putting the identities data set as our first argument and specify our common variable with the `by` argument.
+
+```{r}
+superhero_identities <- superheroes %>%
+  left_join(identities, by = "hero")
+
+superhero_identities
+```
+
+Great, we've joined the data frames with no problem!  For parity, let's also walk through the spreadsheet approach and utilize a VLOOKUP to bring together these data.
```

Comment:
Would suggest the same inversion as above: spreadsheet approach, then R.

Reply:
Agreed but I think it might make sense to show the first vlookup and then mention that its a manual process to pull in each variable. Other wise we extent the article by showing it for each variable we pull in from the other sheet.

Reply:
Wait, just checked this, not sure what you mean exactly as there's a screenshot of the vlookup approach in Excel. Would adding the formula in the article itself make sense?

Reply:
When I say "inversion", I mean show the spreadsheet approach, then the R approach, in general. Rather than the other way around.

Reply:
My only concern with the inversion is that when we have the spreadsheet version last the reader gets the screenshot of the full data frame or output which I think could be more impactful to leave off on.

Reply:
I've updated so we have the excel version first except for the last join. I think the flow of story works in that we're saying oh hey our superheroes data has been updated and we finish that through using the tidyverse and then show the final joined data set using the last VLOOKUP